src/main/java/pl/tomaszdziurko/jvm_bloggers/blogs/UrlBloggersDataFetcher.java

```diff
@@ -0,0 +1,70 @@
+package pl.tomaszdziurko.jvm_bloggers.blogs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import pl.tomaszdziurko.jvm_bloggers.blogs.domain.BlogType;
+import pl.tomaszdziurko.jvm_bloggers.blogs.json_data.BloggersData;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Optional;
+
+import static pl.tomaszdziurko.jvm_bloggers.ApplicationProfiles.DEV;
+import static pl.tomaszdziurko.jvm_bloggers.ApplicationProfiles.PRODUCTION;
+import static pl.tomaszdziurko.jvm_bloggers.ApplicationProfiles.STAGE;
+
+@Profile({PRODUCTION, STAGE, DEV})
+@Component
+@Slf4j
+public class UrlBloggersDataFetcher implements BloggersDataFetcher {
+
+    private final Optional<URL> bloggersUrlOptional;
+    private final Optional<URL> companiesUrlOptional;
+    private final BloggersDataUpdater bloggersDataUpdater;
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @Autowired
+    public UrlBloggersDataFetcher(@Value("${bloggers.data.file.url}") String bloggersDataUrlString,
+                                 @Value("${companies.data.file.url}") String companiesDataUrlString,
+                                  BloggersDataUpdater bloggersDataUpdater) {
+        this.bloggersUrlOptional = convertToUrl(bloggersDataUrlString);
+        this.companiesUrlOptional = convertToUrl(companiesDataUrlString);
+        this.bloggersDataUpdater = bloggersDataUpdater;
+    }
+
+    private Optional<URL> convertToUrl(String urlString) {
+        try {
+            return Optional.of(new URL(urlString));
+        } catch (MalformedURLException exception) {
+            log.error("Invalid URL " + urlString);
```

Comment:
I don't have access to Travis detailed unit tests logs but let's see the [console output](https://s3.amazonaws.com/archive.travis-ci.org/jobs/118455016/log.txt):

```
pl.tomaszdziurko.jvm_bloggers.view.admin.moderation.ModerationPageSpec > Should fetch new blogs FAILED
    junit.framework.AssertionFailedError at ModerationPageSpec.groovy:64

pl.tomaszdziurko.jvm_bloggers.view.admin.moderation.ModerationPageSpec > Should fetch new posts FAILED
    junit.framework.AssertionFailedError at ModerationPageSpec.groovy:64
```

We can conclude from above lines there is some problem in  [`navigateToModerationPageAsAdmin()`](https://github.com/tdziurko/jvm-bloggers/pull/178/files#diff-8c6dcd892112dab0304d5f0265c90d5fR64). Is this method really required? I think it's not to be honest. We can use replace `navigateToModerationPageAsAdmin()` invocation in line [34](https://github.com/tdziurko/jvm-bloggers/pull/178/files#diff-8c6dcd892112dab0304d5f0265c90d5fR34) and [44](https://github.com/tdziurko/jvm-bloggers/pull/178/files#diff-8c6dcd892112dab0304d5f0265c90d5fR44) with `tester.startPage(ModerationPage.class)` instead. This will start your test on _moderation page_ already (without the need to navigate to it from _login page_). But first you have to disable authorization by modifying test [`setup()`] (https://github.com/adamdec/jvm-bloggers/blob/issue%23128/src/test/groovy/pl/tomaszdziurko/jvm_bloggers/view/admin/moderation/ModerationPageSpec.groovy#L27) method like this:

```groovy
def setup() {
    ReflectionTestUtils.setField(wicketApplication, "name", null)
    tester = new WicketTester(wicketApplication)
    // disable authorization to be able to access restricted pages without login
    wicketApplication.securitySettings.authorizationStrategy = IAuthorizationStrategy.ALLOW_ALL
}
```